apps/docs-snippets/src/guide/predicates/predicate-custom-transactions.test.ts

```diff
@@ -0,0 +1,78 @@
+import { ScriptTransactionRequest } from 'fuels';
+import { launchTestNode } from 'fuels/test-utils';
+
+import { ConfigurablePinAbi__factory as PredicateFactory } from '../../../test/typegen';
+import type { ConfigurablePinAbiInputs as PredicateInputs } from '../../../test/typegen';
+
+/**
+ * @group node
+ * @group browser
+ */
+describe('Predicate Custom Transactions', () => {
+  it('uses a predicate in a custom transaction', async () => {
+    // #region predicate-custom-transaction
+    // #import { launchTestNode, ScriptTransactionRequest };
+    // #context import { PredicateFactory } from 'path/to/typegen/outputs';
+    // #context import type { PredicateInputs } from 'path/to/typegen/outputs';
+
+    // Environment setup via the test node
+    using launched = await launchTestNode();
+    const {
+      wallets: [sender, receiver],
+      provider,
+    } = launched;
```

Comment:
As I mentioned in the sync, I believe we should keep the node launcher out of the snippet to keep things less abstract. Unless, of course, we want to teach how to use the utility itself.

```suggestion
    using launched = await launchTestNode();

    const {
      wallets: [_sender, _receiver],
      provider: _provider,
    } = launched;

    const SENDER_PVT_KEY = _sender.privateKey;
    const RECEIVER_ADDRESS = _receiver.address;
    const TESTNET_NETWORK_URL = _provider.url;

    // #region predicate-custom-transaction
    // #import { Provider, ScriptTransactionRequest, TESTNET_NETWORK_URL, Wallet } from 'fuels';
    // #import { PredicateFactory } from 'path/to/typegen/outputs';
    // #import type { PredicateInputs } from 'path/to/typegen/outputs';
    // #import { SENDER_PVT_KEY, RECEIVER_ADDRESS } from 'path/to/my/env/file';

    // Setup
    const provider = await Provider.create(TESTNET_NETWORK_URL);
    const sender = Wallet.fromPrivateKey(SENDER_PVT_KEY);
    const receiver = Wallet.fromAddress(RECEIVER_ADDRESS);
```

Reply:
Good point https://github.com/FuelLabs/fuels-ts/pull/2864/commits/9fb82ce6d840a5b3f280eacf6b83e4e8f3221d37